test/xpu_api/numerics/complex.number/complex/types.pass.cpp

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <complex>
+
+// template<class T>
+// class complex
+// {
+// public:
+//   typedef T value_type;
+//   ...
+// };
+
+#include "support/test_complex.h"
+
+template <class T>
+void
+test()
+{
+    typedef dpl::complex<T> C;
+    static_assert((std::is_same<typename C::value_type, T>::value), "");
+}
+
+template <typename EnableDouble, typename EnableLongDouble>
+void
+run_test()
+{
+    test<float>();
+    oneapi::dpl::__internal::__invoke_if(EnableDouble{}, [&]() { test<double>(); });
+    oneapi::dpl::__internal::__invoke_if(EnableLongDouble{}, [&]() { test<long double>(); });
+}
+
+int main(int, char**)
```

Comment:
Actually, so many changes comparing with the original LLVM test. Indeed,

```
diff --git a/test/xpu_api/numerics/complex.number/complex/types.pass.cpp b/test/xpu_api/numerics/complex.number/complex/types.pass.cpp
index 974d02c08..bf9f8de9e 100644
--- a/test/xpu_api/numerics/complex.number/complex/types.pass.cpp
+++ b/test/xpu_api/numerics/complex.number/complex/types.pass.cpp
@@ -16,24 +16,33 @@
 //   ...
 // };
 
-#include <complex>
-#include <type_traits>
-
-#include "test_macros.h"
+#include "support/test_complex.h"
 
 template <class T>
 void
 test()
 {
-    typedef std::complex<T> C;
+    typedef dpl::complex<T> C;
     static_assert((std::is_same<typename C::value_type, T>::value), "");
 }
 
-int main(int, char**)
+template <typename EnableDouble, typename EnableLongDouble>
+void
+run_test()
 {
     test<float>();
-    test<double>();
-    test<long double>();
+    oneapi::dpl::__internal::__invoke_if(EnableDouble{}, [&]() { test<double>(); });
+    oneapi::dpl::__internal::__invoke_if(EnableLongDouble{}, [&]() { test<long double>(); });
+}
+
+int main(int, char**)
+{
+    // Run on host
+    run_test<::std::true_type, ::std::true_type>();
+
+    // Run test in Kernel
+    TestUtils::run_test_in_kernel([&]() { run_test<::std::true_type, ::std::false_type>(); },
+                                  [&]() { run_test<::std::false_type, ::std::false_type>(); });
 
-  return 0;
+    return TestUtils::done();
 }
```

I propose just one line change - a line "main"  signature, like this:

```
..
-  int main(int, char**)
+ ONEDPL_TEST_NUM_MAIN
{
    test<float>();
    test<double>();
    test<long double>();

  return 0;
}
```

Reply:
Agree. Will do.

Reply:
ONEDPL_TEST_NUM_MAIN - done.
But I think that all another diffs are required, because long double type never support in Kernel, double type - sometimes (depends from device)